src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -165,6 +165,12 @@ public override object Invoke(BindingFlags invokeAttr, Binder? binder, object?[]
                 arguments = CheckArguments(ref stackArgs, parameters, binder, invokeAttr, culture, ArgumentTypes);
```

Comment:
Just noticed unrelated bug above: `if ((InvocationFlags & InvocationFlags.RunClassConstructor) != 0)` condition above looks misplaced. It should be after we validate the number of arguments. We should throw exception when somebody tries to invoke class constructor with arguments.